lms/templates/student_account/account_settings.html

```diff
@@ -27,12 +32,17 @@
 
 <%block name="js_extra">
 <%static:require_module module_name="js/student_account/views/account_settings_factory" class_name="AccountSettingsFactory">
-    var fieldsData = ${ json.dumps(fields) };
-    var authData = ${ json.dumps(auth) };
-    var platformName = ${json.dumps(static.get_platform_name())};
+    var fieldsData = ${ fields | n, dump_js_escaped_json };
+    var authData = ${ auth | n, dump_js_escaped_json };
+    var platformName = ${ static.get_platform_name() | n, dump_js_escaped_json };
 
     AccountSettingsFactory(
-        fieldsData, authData, '${user_accounts_api_url}', '${user_preferences_api_url}', ${user.id}, platformName
+        fieldsData, 
+        authData,
+        '${ user_accounts_api_url | n, js_escaped_string }',
+        '${ user_preferences_api_url | n, js_escaped_string }',
```

Comment:
I did see mention in the docs that url validation is only required if the value is user-supplied.  I guess that does answer that question.